### PR TITLE
MGMT-19234: Add OpenShift sandboxed containers operator(OSC)

### PR DESCRIPTION
**Pull request by pezhang**
Depends on https://github.com/openshift/assisted-test-infra/pull/2579 and https://github.com/openshift/assisted-service/pull/6990

### CODE REVIEW

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml [hunk 1]
```diff
@@ -1140,6 +1140,42 @@ tests:
...
+      OPENSHIFT_TESTS_IMAGE: tests-4.17
+    env:
+      ASSISTED_CONFIG: |
+        OLM_OPERATORS=osc
+        NUM_MASTERS=3
+        NUM_WORKERS=3
+        MASTER_MEMORY=10240
+        MASTER_CPU=6
+        WORKER_MEMORY=10240
+        WORKER_CPU=6
```
Comment (eifrach) [inline ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:1175]: does the resources run's on both masters and workers?

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml [hunk 2]
```diff
@@ -377,6 +377,42 @@ tests:
...
+  optional: true
+  steps:
+    cluster_profile: packet-assisted
+    dependencies:
+      OPENSHIFT_TESTS_IMAGE: tests-4.17
+    env:
+      ASSISTED_CONFIG: |
+        OLM_OPERATORS=osc
+        NUM_MASTERS=1
+        NUM_WORKERS=0
```
Comment (eifrach) [inline ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml:410]: why 1 master?

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml [hunk 3]
```diff
@@ -377,6 +377,42 @@ tests:
...
+  optional: true
+  steps:
+    cluster_profile: packet-assisted
+    dependencies:
+      OPENSHIFT_TESTS_IMAGE: tests-4.17
+    env:
+      ASSISTED_CONFIG: |
+        OLM_OPERATORS=osc
+        NUM_MASTERS=3
+        NUM_WORKERS=3
```
Comment (eifrach) [inline ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml:391]: no need for 3 workers max 2
this is due to the BM resources.

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml [hunk 4]
```diff
@@ -377,6 +377,42 @@ tests:
...
+        MASTER_MEMORY=10240
+        MASTER_CPU=6
+        WORKER_MEMORY=10240
+        WORKER_CPU=6
```
Comment (eifrach) [inline ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml:395]: same

**Review by eifrach — CHANGES_REQUESTED** [review]

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml [hunk 5]
```diff
@@ -1140,6 +1140,42 @@ tests:
...
+  run_if_changed: ^(internal/operators/.*\.go|internal/operators/osc/.*|internal/operators/api/.*|internal/operators/handler/.*)$
+  steps:
+    cluster_profile: packet-assisted
+    dependencies:
+      OPENSHIFT_TESTS_IMAGE: tests-4.17
+    env:
+      ASSISTED_CONFIG: |
+        OLM_OPERATORS=osc
+        NUM_MASTERS=1
+        NUM_WORKERS=0
```
[the remark this answers is not in the document]
Comment (pezhang) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:1154]: @eifrach Yes, this is for the OSC operator in the SNO cluster. OSC operator work for both SNO and regular clusters.

File: ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml (hunk 1 above)
Comment (pezhang) [reply ci-operator/config/openshift/assisted-service/openshift-assisted-service-master__edge.yaml:1175]: @eifrach Yes. I think so.

**Comment by pezhang** [comment]
Hello @eifrach ,  I've replied your comments in line and changed the worker number to 2, does that make sense? Do you have further comments? And do I need to test this PR locally? If yes, how to test it? Thanks a lot.

File: ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml (hunk 2 above)
Comment (eifrach) [reply ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml:410]: so can you add `sno` to the name of the test ? `e2e-metal-assisted-osc-sno-4-17`

Comment (pezhang) [reply ci-operator/config/openshift/assisted-test-infra/openshift-assisted-test-infra-master.yaml:410]: @eifrach Updated done. Thanks.

**Comment by pezhang** [comment]
needs-ok-to-test

**Pull request merged**